File: src/main/java/com/hosigus/tools/options/NetOption.java
package com.hosigus.tools.options;

import com.hosigus.tools.items.JSONBean;

import java.util.HashMap;
import java.util.Map;

/**
 * Created by 某只机智 on 2018/5/7.
 */

public class NetOption {
    private String url;
    private Map<String, String> paramSet;
    private Class<? extends JSONBean> beanClass;


    public NetOption(String url) {
        this.url = url;
    }

    public NetOption param(String key, String value) {
        if (paramSet == null) {
            paramSet = new HashMap<>();
        }
        paramSet.put(key, value);
        return this;
    }

    public NetOption beanClass(Class<? extends JSONBean> beanClass) {
        this.beanClass = beanClass;
        return this;
    }

    public String getUrl() {
        return url;
    }

    public Map<String, String> getParamSet() {
        return paramSet;
    }

    public Class<? extends JSONBean> getBeanClass() {
        return beanClass;
    }
}

File: src/main/java/com/hosigus/tools/options/NetOption.kt
package com.hosigus.tools.options

import com.hosigus.tools.items.JSONBean
import kotlin.collections.HashMap

/**
 * Created by Hosigus on 2018/6/5.
 */
class NetOption(val url: String) {
    var paramSet: MutableMap<String, String>? = null
        private set
    var headSet: MutableMap<String, String>? = null
        private set
    var beanClass: Class<out JSONBean>? = null
        private set

    fun param(key: String, value: String): NetOption {
        if (paramSet == null) {
            paramSet = hashMapOf()
        }
        paramSet!![key] = value
        return this
    }

    fun head(key: String, value: String): NetOption {
        if (headSet == null) {
            headSet = hashMapOf()
        }
        headSet!![key] = value
        return this
    }

    fun beanClass(beanClass: Class<out JSONBean>): NetOption {
        this.beanClass = beanClass
        return this
    }

}